WalletWasabi.Fluent/ViewModels/Wallets/Send/CurrencyConversion/CurrencyInputViewModel.cs

```diff
@@ -0,0 +1,709 @@
+using ReactiveUI;
+using System.Linq;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using WalletWasabi.Fluent.Helpers;
+using WalletWasabi.Fluent.Models.Currency;
+using WalletWasabi.Fluent.Models.UI;
+using WalletWasabi.Fluent.Models.Wallets;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Fluent.ViewModels.Wallets.Send.CurrencyConversion;
+
+public partial class CurrencyInputViewModel : ActivatableViewModel
+{
+	public const string DecimalSeparator = ".";
+
+	[AutoNotify] private decimal _maxValue;
+	[AutoNotify] private decimal _minSuggestionValue;
+	[AutoNotify] private string _text = "";
+	[AutoNotify] private string _formattedText = "";
+	[AutoNotify] private int? _decimalSeparatorPosition;
+	[AutoNotify] private int? _integralLength;
+	[AutoNotify] private int? _fractionalLength;
+	[AutoNotify] private CurrencyValue _value = CurrencyValue.EmptyValue;
+
+	public CurrencyInputViewModel(UiContext uiContext, CurrencyFormat currencyFormat, IWalletModel? wallet = null, bool enableClipboardSuggestion = false)
+	{
+		CurrencyFormat = currencyFormat;
+
+		if (wallet is { } && enableClipboardSuggestion)
+		{
+			Suggestion = new CurrencyInputClipboardListener(uiContext, wallet, this);
+		}
+	}
+
+	public CurrencyFormat CurrencyFormat { get; }
+
+	public CurrencyInputClipboardListener? Suggestion { get; }
+
+	public int InsertPosition { get; private set; }
+	public int? SelectionStart { get; set; }
+	public int? SelectionEnd { get; set; }
+
+	public bool HasSelection => SelectionEnd is { } && SelectionEnd is { };
```

Comment:
One here should be `SelectionStart` right?

Reply:
Fixed